src/Config/Validator/Build/BalerSupport.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Validator\Build;
+
+use Magento\MagentoCloud\Config\Magento\Shared\ReaderInterface;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+use Symfony\Component\Process\ExecutableFinder;
+
+/**
+ * Verify that the build configuration is ready for baler.
+ */
+class BalerSupport implements ValidatorInterface
+{
+    /**
+     * @var ExecutableFinder
+     */
+    private $finder;
+
+    /**
+     * @var Validator\ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @var ReaderInterface
+     */
+    private $configReader;
+
+    /**
+     * @param Validator\ResultFactory $resultFactory
+     * @param ExecutableFinder $finder
+     * @param ReaderInterface $configReader
+     */
+    public function __construct(
+        Validator\ResultFactory $resultFactory,
+        ExecutableFinder $finder,
+        ReaderInterface $configReader
+    ) {
+        $this->resultFactory = $resultFactory;
+        $this->finder = $finder;
+        $this->configReader = $configReader;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        $errors = [];
+
+        if ($this->finder->find('baler') === null) {
+            $errors[] = 'Path to baler executable could not be found.'
+                . ' The Node package may not be installed or may not be linked.';
+        }
+
+        $magentoConfig = $this->configReader->read();
```

Comment:
You are reading configuration only from `config.php` file but it can be stored in the database or env variables as well. For modules it's ok, but for system config, you should use `bin/magento config:show` command.

Reply:
Will `config:show` work without the DB though? One of the agreements Jeff and I came to was that we would only support running Baler during build, not the deployment phase, as an added incentive for users to do SCD on build.

Reply:
I think no, my bad I forgot that we don't have access to the database on the Build phase.
So this functionality requires to have dumped configuration into config.php.